### ISSUE TITLE
Use JPype instead of subprocess

### ISSUE THREAD

**Issue by mara004** [chezou/tabula-py#352]
**Is your feature request related to a problem? Please describe.**
Repeatedly calling the Jar as java subprocess causes overhead (start java, load modules, etc.)

**Describe the solution you'd like**
Switch to JPype to launch the VM only once and call into the APIs directly.
For a start, the CLI main functions could be used to simplify the transition, maybe switching to the actual API level eventually.

**Describe alternatives you've considered**
Sticking with subprocess is possible but has the drawbacks described above.

**Additional context**
The python-pdfbox project had a similar transition from subprocess to JPype.

**Comment by chezou** [comment]
Thanks for suggestion. Initially I thought to use py4j, but at that time, tabula-java development was active, so I pended the idea for maintainability.

I'd think tweaking `io._run` function and `TabulaOption` could help to implement it. https://github.com/chezou/tabula-py/blob/b24e3bd9d0f0059fdb06a04bb561ae04efe8dce4/tabula/io.py#L58-L102

Or, we may be able to do something similar as tabula-rs https://github.com/tabulapdf/tabula-java/pull/444

Anyway, I'm looking forward to contribution for the change.

**Comment by mara004** [comment]
Thanks for the considerations!

> Anyway, I'm looking forward to contribution for the change.

Disclaimer: Unfortunately, I'm fairly busy with some other projects (pypdfium2 & co.) and thus don't have an intent to work on this myself.

**Comment by chezou** [comment]
I tried jpype on https://github.com/chezou/tabula-py/pull/355

However, I encountered when I run pytest with multiple files, it always fails.
https://github.com/chezou/tabula-py/actions/runs/5987819184

This is a huge blocker for introducing jpype, and I'm about to give up.

**Comment by mara004** [comment]
I'm still a bit tired, but all the test suite failures seem to be caused by `TypeError: _run() takes from 2 to 3 positional arguments but 4 were given`.
That's because the `encoding` argument was removed. Assuming that is correct, just adapt the calling code accordingly. Please don't give up just yet, and at least leave the PR open as draft so it is more visible for others to take a look.

**Comment by chezou** [comment]
I guess you're trying the old PR. Can you check thenew one? https://github.com/chezou/tabula-py/pull/356

**Comment by mara004** [comment]
I don't currently have the env to test this locally, I can only try to help with concrete questions.
The benchmark you shared looks promising. Any problems left?

**Comment by chezou** [comment]
Ah, sorry for the confusion. I solved the issue with separating test files and processes. https://github.com/chezou/tabula-py/pull/356#discussion_r1306600161

It's okay to merge into master for now.

**Comment by chezou** [comment]
Found a weird error blames of jpype. I can't reproduce it, so need to get some help.
https://stackoverflow.com/questions/77077943/pyspark-tabula-py-read-pdf-error-no-module-named-org-apache-commons/77092171#77092171

**Comment by chezou** [comment]
It presumably is a specific environment issue with jpype.
https://community.databricks.com/t5/data-engineering/what-is-the-best-way-to-read-in-a-ms-access-accdb-database-into/td-p/3217

I will implement a workaround to enable subprocess as an option.

**Comment by mara004** [comment]
Sorry for the problems, I'm afraid I don't know the cause either.
I'd suggest you ask about this upstream on jpype's discussion page, maybe they have seen this already.

**Comment by chezou** [comment]
@mara004 Thanks for your suggestion. However, I can't reproduce issue by my side and I'm concerned to raise on behalf of them since I don't know what is the trigger.

I released v2.8.2 to automatically fallback to subprocess if there's any import error on jpype.
